### DISCUSSION TITLE
1.30.0

### DISCUSSION THREAD

**Discussion by dani-garcia** [dani-garcia/vaultwarden#4034] [General]
⚠️ **Note:** The WebSockets service for live sync has been integrated in the main HTTP server, which means simpler proxy setups that don't require a separate rule to redirect WS traffic to port 3012. Please check the updated examples in the [wiki](https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples). It's recommended to migrate to this new setup as using the old server on port 3012 is deprecated, won't receive new features and will be removed in a future release.

## Major changes and New Features
* Added `passkey` support, allowing the browser extensions to store and use your `passkeys`, make sure the extension is updated to version `2023.10.0` or newer for passkey support.
* Updated web vault to 2023.10.0.
* Fixed crashes in ARMv6 devices
* Fixed crashes when trying to create/edit a cipher in the mobile applications.

## What's Changed
* Update Rust and Crates by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/3808
* update web-vault to v2023.8.2 by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/3821
* Fix Login With Device without MasterPassword by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/3831
* Update GitHub Workflow by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/3910
* Fix arm builds by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/3911
* Fix typos by @tuhanayim in https://github.com/dani-garcia/vaultwarden/pull/3959
* csp: rename anonaddy.com to addy.io by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/3950
* filter handlebars logs by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/3859
* Remove unnecessary variable clone by @mvalois in https://github.com/dani-garcia/vaultwarden/pull/3981
* README.md: Fix grammar nit by @AndreasHGK in https://github.com/dani-garcia/vaultwarden/pull/3965
* Fix small issues by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/3964
* Adds LastActive on /admin/users API route by @mvalois in https://github.com/dani-garcia/vaultwarden/pull/3951
* Reopen log file on SIGHUP by @tobiasmboelz in https://github.com/dani-garcia/vaultwarden/pull/3909
* Fix External ID not set during DC Sync by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/3804
* New config option disable email change by @admav in https://github.com/dani-garcia/vaultwarden/pull/3986
* 2FA Confirmation Code Email subject line change to fix triggering Google spam blocker by @aureateflux in https://github.com/dani-garcia/vaultwarden/pull/3572
* Implement cipher key encryption by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/3990
* Container building changes by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/3958
* Fix issue with MariaDB/MySQL migrations by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/3994
* feat: Working passkeys storage by @GeekCornerGH in https://github.com/dani-garcia/vaultwarden/pull/4025
* ci: add trivy workflow by @mightyBroccoli in https://github.com/dani-garcia/vaultwarden/pull/3997
* Fix importing Bitwarden exports by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4030

## New Contributors
* @tuhanayim made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/3959
* @mvalois made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/3981
* @AndreasHGK made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/3965
* @tobiasmboelz made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/3909
* @admav made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/3986
* @aureateflux made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/3572
* @mightyBroccoli made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/3997

**Full Changelog**: https://github.com/dani-garcia/vaultwarden/compare/1.29.2...1.30.0

<hr /><em>This discussion was created from the release <a href='https://github.com/dani-garcia/vaultwarden/releases/tag/1.30.0'>1.30.0</a>.</em>

**Comment by alexdelprete** [comment]
Upgraded to 1.30.0 yesterday. Tested passkey creation on google account site: VW saves the passkey correctly, but when logging in, it doesn't find it.

I have browser extension 2023.10.1, is the server version correct for VW 1.30.0?:

![image](https://github.com/dani-garcia/vaultwarden/assets/7027842/f867bc91-84dc-426d-a2eb-9f77d963ebb6)

**Reply by BlackDex** [reply]
We report the least compatible version there to make sure the clients work. We will not update that everytime new clients will be released.

**Reply by alexdelprete** [reply]
the "least" compatible? shouldn't be the "most" compatible (older) version?

**Reply by BlackDex** [reply]
Ill rephrase, the lowest version compatible so that the clients work without issues.

**Reply by alexdelprete** [reply]
Got it.

Anyway, I also tested with https://www.passkeys.io/ and https://webauthntest.identitystandards.io/: it works beautifully.

So it's a specific issue with Google I guess.

**Reply by leo15dev** [reply]
https://github.com/bitwarden/clients/issues/6764
Just wait for the bitwarden engineering team to fix it.

**Reply by alexdelprete** [reply]
Thank you.

**Reply by leo15dev** [reply]
Browser v2023.10.2 https://github.com/bitwarden/clients/releases/tag/browser-v2023.10.2

Fixes to iCloud and Google passkey login. There is still a known issue with Google on some Windows device configurations.

**Reply by alexdelprete** [reply]
Thanks. Extensions update still not available, I guess it will take some time for vendors to publish it. How long does it take generally, days/weeks?

**Reply by BlackDex** [reply]
For Firefox it's released. Not sure which browser you use. But i have no clue to be honest.

**Reply by alexdelprete** [reply]
I use Thorium/Chrome/Edge. No updates yet.

I checked Firefox: have 10.1 and I checked updates...nothing found. The [link](https://addons.mozilla.org/en-US/firefox/addon/bitwarden-password-manager/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search) to the store confirms it's still 10.1.

![image](https://github.com/dani-garcia/vaultwarden/assets/7027842/3bc93ae5-dbef-4385-9bd5-dbe20a86d4b2)

**Reply by BlackDex** [reply]
That is then correct version. v2023.10.x has passkey support for browsers.

**Reply by alexdelprete** [reply]
> That is then correct version. v2023.10.x has passkey support for browsers.

The bug which I posted about (google) is solved in 2023.10.2: https://github.com/bitwarden/clients/releases/tag/browser-v2023.10.2

**Reply by alexdelprete** [reply]
I'm referring to #4038

**Comment by logopk** [comment]
Oh I see....


Yet the display is not consistent - isn't it?

**Reply by BlackDex** [reply]
The display is a bit weird. We used to put the Vaultwarden version there, but since clients use a version check to enable or disable some features we need to provide a Bitwarden compatible version there.

**Reply by alexdelprete** [reply]
So the version provided there is actually the equivalent, funcionality-wise, of BW server?

**Reply by BlackDex** [reply]
Sorta yes. But Bitwarden only checks currently if the version of the server is greater then or equal to 2023.9.1. since we do not know if we support all features of the 2023.10.x version, we need to keep on the lower side.

**Reply by alexdelprete** [reply]
Makes sense. Thanks for the explanation.

**Comment by NoseyNick** [comment]
Wondering if this was expected to have made it into the `vaultwarden/server:alpine` docker image yet? ... or did something go wrong with the docker build/release? Or perhaps we are supposed to know this is a `vaultwarden/server:testing-alpine` image "cos it's a .0 release" or something?
Much appreciated!   :beer: :+1:

**Reply by BlackDex** [reply]
Use `latest-alpine`, there was a bug which didn't updated the `alpine` tag.

**Reply by NoseyNick** [reply]
Works, many thanks!   :star: :beer: :+1:
